Include/pyport.h

```diff
@@ -607,4 +607,21 @@ extern "C" {
 #  define _SGI_MP_SOURCE
 #endif
 
+// Explicit fallthrough in switch case to avoid warnings
+// with compiler flag -Wimplicit-fallthrough.
+//
+// Usage example:
+//
+//     switch (value)
+//     case 1: _Py_FALLTHROUGH;
+//     case 2: code; break;
+//     }
+//
+// __attribute__((fallthrough)) was introduced in GCC 7.
+#if defined(__has_attribute) && __has_attribute(fallthrough)
```

Comment:
it looks like there are many compiler warnings here on Windows:

```
unexpected tokens following preprocessor directive - expected a newline [D:\a\cpython\cpython\PCbuild\_freeze_module.vcxproj]
```

Reply:
Yeah I worried about this issue.

Reply:
I rewrote the fix to avoid preprocessor issue with __has_attribute(). Instead, I added _Py__has_attribute() macro.